data_structures/arrays/permutations.py

```diff
@@ -1,33 +1,32 @@
-def permute(nums: list[int]) -> list[list[int]]:
+def permute_iterative(nums: list[int]) -> list[list[int]]:
     """
     Return all permutations.
-    >>> from itertools import permutations
-    >>> numbers= [1,2,3]
-    >>> all(list(nums) in permute(numbers) for nums in permutations(numbers))
-    True
+
+    >>> permute_backtrack([1, 2, 3])
+    [[1, 2, 3], [1, 3, 2], [2, 1, 3], [2, 3, 1], [3, 2, 1], [3, 1, 2]]
     """
-    result = []
+    result: list[list[int]] = []
     if len(nums) == 1:
         return [nums.copy()]
```

Comment:
```suggestion
    if len(nums) == 0:
        return [[]]
```
I think we should make the base case 0 rather than 1. This slightly extends the function to work on an empty list input, and this matches the behavior of `itertools.permutations`.